### ISSUE TITLE
Add Softmax layer

### ISSUE THREAD

**Issue by zaleslaw** [JetBrains/KotlinDL#52]
We are missing some activation layers to support the export of models from Keras fully. One of them is the Softmax layer.

Add an activation layer class, write documentation for it, write a test for it, try, if possible, create a small trainable network with it (in your own GitHub) and attach a link here in the comments.

The layer should be placed [here](https://github.com/JetBrains/KotlinDL/tree/master/api/src/main/kotlin/org/jetbrains/kotlinx/dl/api/core/layer/activation)

As a reference implementation, the [ReLU activation layer](https://github.com/JetBrains/KotlinDL/blob/master/api/src/main/kotlin/org/jetbrains/kotlinx/dl/api/core/layer/activation/ReLU.kt) could be used, but feel free to improve it!

Also, support for export and import of activation layer in JSON format should be added (see [ModelLoader.kt ](https://github.com/JetBrains/KotlinDL/blob/master/api/src/main/kotlin/org/jetbrains/kotlinx/dl/api/inference/keras/ModelLoader.kt) and [ModelSaver.kt](https://github.com/JetBrains/KotlinDL/blob/master/api/src/main/kotlin/org/jetbrains/kotlinx/dl/api/inference/keras/ModelSaver.kt))

A detailed description of the activation layer can be found [here](https://keras.io/api/layers/activation_layers/softmax/)

**Comment by d-lowl** [comment]
I'm currently looking into this issue and have a question regarded the expected implementation. TF/Keras Softmax layer in Python takes two arguments: the input tensor and the mask. However, the current abstract forward function does not allow that. Do we want to implement masking (and hence extending the signature for `forward`) or is it out of scope for this issue?

**Comment by zaleslaw** [comment]
@d-lowl It's a reasonable question: so, I suppose we can implement this layer without a mask.

As you rightly noted, at the moment there is no support for masks, including at the level of layers and arguments. The fact is that while they are not needed for those models that are supported in the model zoo. I have a secret hope that they may not be needed.

**Comment by d-lowl** [comment]
Gotcha. I'll hopefully come back with a PR soon.

**Comment by zaleslaw** [comment]
So, I've assign it for you, if you are agree with that, to avoid two or more PRs for one tickets
